internal/commands/app/new_app_inputs.go

```diff
@@ -62,3 +65,56 @@ func (i *newAppInputs) resolveRemoteApp(ui terminal.UI, rc realm.Client) (realm.
 	}
 	return ra, nil
 }
+
+func (i *newAppInputs) resolveTemplateID(ui terminal.UI, client realm.Client) (string, error) {
+	if i.Template == "" && ui.AutoConfirm() {
+		return "", nil
+	}
+
+	templates, err := client.Templates()
+	if err != nil {
+		return "", err
+	}
+
+	// do not disrupt application creation flow if templates are not
+	// available and user is not specifying a template
+	if i.Template == "" && len(templates) == 0 {
+		return "", nil
+	}
+
+	if len(templates) == 0 {
```

Comment:
> Without the --use-template flag and when the /templates endpoint doesn't return anything, the user cannot create any apps because it'll stop here since we currently always prompt the user for the template id

I reeeeally want to avoid ever implementing something like `--use-template`, because it's such a single-use escape hatch feature... the only time you would ever use it would be to set `--use-template false`, which feels extremely clunky to me.

here, we only error out if we failed to fetch templates _and_ a user has specified a template param. if they don't specify a template, we return `""` above on line 82, allowing the user to continue as usual and create non-template apps. if we don't have any templates, there's really no way forward for creating a templated app, but they can still create a regular "blank" app.

Reply:
ah okay yeah i see now. I'm good with this then - feel free to ignore!